stats/opentelemetry/e2e_test.go

```diff
@@ -582,3 +610,625 @@ func pollForWantMetrics(ctx context.Context, t *testing.T, reader *metric.Manual
 
 	return fmt.Errorf("error waiting for metrics %v: %v", wantMetrics, ctx.Err())
 }
+
+// TestServerWithMetricsAndTraceOptions tests emitted metrics and traces from
+// OpenTelemetry instrumentation component. It then configures a system with a gRPC
+// Client and gRPC server with the OpenTelemetry Dial and Server Option configured
+// specifying all the metrics and traces provided by this package, and makes a Unary
+// RPC and a Streaming RPC. These two RPCs should cause certain recording for each
+// registered metric observed through a Manual Metrics Reader on the provided
+// OpenTelemetry SDK's Meter Provider. It also verifies the traces are recorded
+// correctly.
+func (s) TestServerWithMetricsAndTraceOptions(t *testing.T) {
```

Comment:
actually, why do we need this test separately? As we discussed offline we can make the TestSpan test do both unary and streaming rpc and verify everything. Similar to what is done in open census test. Then we can delete this test.

Reply:
This test is specially to verify that the new feature works as expected with metrics option as well. We do have `TestSpan` test which works similar to what we have for open census package. Should we merge them in a single `TestSpan` test?

Reply:
what is new feature

Reply:
I was referring to otel tracing.